core/unit_test/TestRange.hpp

```diff
@@ -60,33 +60,10 @@ struct TestRange {
     Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace, ScheduleType>(0, N),
                          *this);
 
-    {
-      using ThisType = TestRange<ExecSpace, ScheduleType>;
-      std::string label("parallel_for");
-      Kokkos::Impl::ParallelConstructName<ThisType, void> pcn(label);
-      ASSERT_EQ(pcn.get(), label);
-      std::string empty_label("");
-      Kokkos::Impl::ParallelConstructName<ThisType, void> empty_pcn(
-          empty_label);
-      ASSERT_EQ(empty_pcn.get(), typeid(ThisType).name());
-    }
-
```

Comment:
Why delete these ones (instead of e.g. moving them). I mean sure we test the end result but in principle I wouldn't mind having actual unit tests for implementation details too,

Reply:
It is an `Impl::` facility, it does not guarantee that it is properly called in our parallel constructs.

Reply:
Yeah obviously, but my point is having it too tells you whether the name generation failed in some way, or whether the passing on failed.

Reply:
Re-added them into the new test file